db/database.go

```diff
@@ -64,8 +69,16 @@ func (db *Database) GetAllOffers() ([]*types.Offer, error) {
 
 	offers := []*types.Offer{}
 	for iter.Valid() {
+		key := iter.Key()
+
+		// if the key becomes longer than 32, we're not iterating over offers
+		if len(key) > 32 {
+			break
+		}
```

Comment:
the underlying iterator is a bit messed up, it returns everything in the db starting at keys starting with `offer` and only slices off the prefix for those. I'll work on fixing that/making a workaround cause I agree this is cursed

and yeah, I think it could be reduced to 16 bytes, that leaves 2^128 different offer IDs which is more than enough